src/groups/mqb/mqbplug/mqbplug_authenticator.h

```diff
@@ -173,53 +164,42 @@ class AuthenticatorPluginFactory : public PluginFactory {
     create(bslma::Allocator* allocator) = 0;
 };
 
-// ============================================================================
-//                             INLINE DEFINITIONS
-// ============================================================================
-
-// --------------------------
-// class AuthenticationResult
-// --------------------------
-
-// CREATORS
-
-inline AuthenticationResult::AuthenticationResult(bslma::Allocator* allocator)
-: d_principal(allocator)
-, d_lifetimeMs()
-{
-    // NOTHING
-}
+// ========================
+// struct AuthenticatorUtil
+// ========================
 
-// ACCESSORS
+struct AuthenticatorUtil {
+    // STATIC CLASS METHODS
 
-inline const bsl::string& AuthenticationResult::principal() const
-{
-    return d_principal;
-}
+    /// Find authenticator config with the specified `name`
+    static const mqbcfg::AuthenticatorPluginConfig*
+    findAuthenticatorConfig(bsl::string_view name);
+};
 
-inline const bsl::optional<bsls::Types::Int64>&
-AuthenticationResult::lifetimeMs() const
-{
-    return d_lifetimeMs;
-}
+// ============================================================================
+//                             INLINE DEFINITIONS
+// ============================================================================
 
-// MANIPULATORS
+// ------------------------
+// class AuthenticationData
+// ------------------------
 
-inline void AuthenticationResult::reset()
+inline AuthenticationData::AuthenticationData(
+    const bsl::vector<char>& authnPayload,
+    bsl::string_view         clientIpAddress)
+: d_authnPayload(authnPayload)
+, d_clientIpAddress(clientIpAddress)
```

Comment:
```suggestion
, d_clientIpAddress(clientIpAddress, allocator)
```
Missing allocator here